### PR TITLE
REF list teams by slug and remove members/teams properly

### LINKED ISSUE #1370: search for teams by handle/slug and not name
I think this is a simple as using `team.slug` instead of `team.name` in the relevant code.

cc @isuruf

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

closes #1370

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/core This PR is ready for review. 

Before we merge, we need to figure out the error from last time. @jakirkham @isuruf @pelson any thoughts here?

File: conda_smithy/github.py [hunk 1]
```diff
@@ -220,21 +220,21 @@ def configure_github_team(meta, gh_repo, org, feedstock_name):
     # Try to get team or create it if it doesn't exist.
     team_name = feedstock_name
     current_maintainer_teams = list(gh_repo.get_teams())
-    team = next(
+    fs_team = next(
```
Comment (isuruf) [inline conda_smithy/github.py:223]: why?

Comment (beckermr) [reply conda_smithy/github.py:223]: The `team` variable is used inside a loop below. I made this change to try and make the code clearer and more explicit.

**Comment by beckermr** [comment]
Bump here @jakirkham. No worries if you don't recall.

**Comment by jakirkham** [comment]
Sorry, no clue. This is something Phil told me about and not something I ever saw in practice 🤷‍♂️

**Comment by jakirkham** [comment]
One option as far as testing goes, would be to restrict this to a specific team and make it a no-op otherwise. That way we could play with adding/removing from that team until we get some confidence that this is working. Then we can relax the team restriction.

**Comment by beckermr** [comment]
Already did this and tested the code here: https://github.com/beckermr/misc/blob/master/work/conda_forge_test_teams_mems_remove/test_teams_mems.py

**Comment by pelson** [comment]
Glad to see this get some love :heart:
I started getting emails from GitHub a few weeks ago that removed me from a few teams. About 5000 of them :joy:.
Anyway, this change looks good :+1:. Just a word of warning, in the early days of conda-forge I once tweaked this script and got it wrong. It ended up notifying **all** of the contributors and removing them from **all** of the teams. At the time that was like 50 contributors and ~300 repos, so compared to if that happened now, wasn't a big deal. That is why the script had the remove part disabled - I intended to just go through the logs and remove the few numbers of people/teams manually, but obviously the automated approach is so much more desirable (its the conda-forge way :smile:).

**Comment by beckermr** [comment]
Thanks @pelson! I almost removed all feedstocks from appveyor a while back. It's definitely a risk we take when doing this work!

**Pull request merged**